src/texture-processor/image.hh

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include <texture-processor/image_view.hh>
+#include <texture-processor/storage.hh>
+
+namespace tp
+{
+/**
+ * An image is an image_view with an associated storage
+ * Basically the value-semantics owning version of image_view
+ * It can also be thought of as a cc::vector but with an image API
+ *
+ * TODO:
+ *  - how are non-natural strides handled in creation/resize?
+ */
+template <class BaseTraits>
+struct image : image_view<BaseTraits>
+{
+    using traits = tp::traits<BaseTraits>;
+    using pixel_t = typename traits::pixel_t;
+    using pos_t = typename traits::pos_t;
+    using ipos_t = typename traits::ipos_t;
+    using extent_t = typename traits::extent_t;
+    using storage_t = typename traits::storage_t;
+    using storage_view_t = typename traits::storage_view_t;
+    using data_ptr_t = typename storage_view_t::data_ptr_t;
```

Comment:
Think it would be nice to have a different solution for this, or at least explain each one better up front